File: common/src/main/java/org/tdf/common/store/MemoryDatabaseStore.java
package org.tdf.common.store;

import java.util.Iterator;
import java.util.Map;

public class MemoryDatabaseStore extends ByteArrayMapStore<byte[]> implements DatabaseStore {
    public MemoryDatabaseStore() {
    }

    public MemoryDatabaseStore(Map<byte[], byte[]> map) {
        super(map);
    }

    @Override
    public void init(DBSettings settings) {

    }

    @Override
    public boolean isAlive() {
        return true;
    }

    @Override
    public void close() {

    }
}

File: common/src/main/java/org/tdf/common/store/MemoryDatabaseStore.kt
package org.tdf.common.store


class MemoryDatabaseStore : ByteArrayMapStore<ByteArray>(), DatabaseStore {
    override fun init(settings: DBSettings) {}
    override val isAlive: Boolean get() {
        return true
    }

    override fun close() {}
    override fun clear() {
        cache.clear()
    }
}